src/amberscript/parser.cc

```diff
@@ -1368,6 +1513,9 @@ Result Parser::ParseValues(const std::string& name,
 
       v.SetIntValue(token->AsUint64());
     }
+    ++seg_idx;
```

Comment:
Just to confirm, vec4<float> is 4 segments? How are runtime arrays to be handled?

Reply:
A vec4 would be 4 segments, yes. My plan for runtime arrays is that you end up looping over that group of segments again. So, the start of the array will be flagged and you can have 0 or more of them as needed.